phoenix-core/src/main/java/org/apache/phoenix/compile/WhereCompiler.java

```diff
@@ -200,6 +201,15 @@ protected ColumnRef resolveColumn(ColumnParseNode node) throws SQLException {
                 return ref;
             }
             PTable table = ref.getTable();
+            // If current table in the context is local index and table in column reference is global that
+            // means the column is not present in the local index. Local indexes do not currently support this,
+            // so we need skip this plan.
+            if (context.getCurrentTable().getTable().getIndexType() == IndexType.LOCAL
+                    && (table.getIndexType() == null || table.getIndexType() == IndexType.GLOBAL)) {
+                String schemaNameStr = table.getSchemaName()==null?null:table.getSchemaName().getString();
+                String tableNameStr = table.getTableName()==null?null:table.getTableName().getString();
+                throw new ColumnNotFoundException(schemaNameStr, tableNameStr, null, ref.getColumn().getName().getString());
```

Comment:
It is not clear how returning ColumnNotFoundException results in skipping a plan. Can we add some comments here for that?

Reply:
I just brought that code back from before PHOENIX-5109.
It throws an exception, which is then caught in query enumeration in the optimizing phase and consequently ignores that plan.
Happy to add a comment to that extent.

Reply:
Something like, "This exception will be caught in query enumeration in the optimizing phase and consequently the plan will be ignored" as you wrote, will be helpful. I will then approve it. Thanks!

Reply:
How about this?

Reply:
LGTM